### ISSUE TITLE
Cannot read property 'name' of undefined

### ISSUE THREAD

**Issue by franciscolourenco** [steelbrain/php-serialize#258]
Node `querystring.parse` method returns an object with null prototype for security reasons.

Passing that object to serialize() throws `Cannot read property 'name' of undefined`.

Would it be possible to stop depending on the object's prototype in order to fix this issue?

Thanks!

**Comment by philipp-tailor** [comment]
👍🏼  for this request.

Is the library still actively maintained?

**Comment by franciscolourenco** [comment]
https://nodejs.org/api/querystring.html#querystring_querystring_parse_str_sep_eq_options

**Comment by steelbrain** [comment]
> Is the library still actively maintained?

Yes. The bug seems to be out of scope of the lib tho, you could just add an if/else statement in your own code to handle this

**Comment by franciscolourenco** [comment]
@steelbrain A common way to parse `url-encoded-form` data is to use the native node [querystring.parse()](https://nodejs.org/api/querystring.html#querystring_querystring_parse_str_sep_eq_options) function. Since this is a node library it would probably be a good idea to be compatible with node generally speaking.

Is there anything which prevents php-serialize form using a more fail-proof method of identifying objects, like for instance: 

```
typeof object === 'object' && object !== null
```
